internal/query/access_token.go

```diff
@@ -107,11 +107,11 @@ func (q *Queries) ActiveAccessTokenByToken(ctx context.Context, token string) (m
 	ctx, span := tracing.NewSpan(ctx)
 	defer func() { span.EndWithError(err) }()
 
-	split := strings.Split(token, "-")
-	if len(split) != 2 {
-		return nil, zerrors.ThrowUnauthenticated(nil, "QUERY-LJK2W", "Errors.OIDCSession.Token.Invalid")
+	sessionToken, err := command.DecodeOIDCSessionToken(token)
+	if err != nil || sessionToken.ID == "" || sessionToken.AccessToken == "" {
+		return nil, zerrors.ThrowUnauthenticated(err, "QUERY-LJK2W", "Errors.OIDCSession.Token.Invalid")
 	}
```

Comment:
Why do we need to do decoding steps at this stage of the business logic? This can ll be handled in the middleware.